pyroaring/abstract_bitmap.pxi

```diff
@@ -194,8 +194,57 @@ cdef class AbstractBitMap:
         return str(self)
 
     def __str__(self):
-        values = ', '.join([str(n) for n in self])
-        return '%s([%s])' % (self.__class__.__name__, values)
+        skip_rows = len(self) > 500 #this is the cutoff number for the truncating to kick in.
+        table_max_width = 80  # this isn't the length of the entire output, it's only for the numeric part
+        num_lines_if_skipping = 5  # the number of lines to show in the beginning and the end when output is being truncated
+
+        head = self.__class__.__name__ + '(['
+        row_start_buffer = ' ' * len(head)
+        tail = '])'
+
+        try:
+            maxval = self.max()
+        except ValueError:
+            # empty bitmap
+            return head + tail
+
+        element_max_length = len(str(maxval))
+        column_width = element_max_length + 2
+
+        num_columns = table_max_width // column_width
+
+        num_rows = len(self) / num_columns
+        if not num_rows.is_integer():
```

Comment:
In Python2, this division is an integer division. Convert one of the operands to `float` to avoid this:
```python
num_rows = len(self) / float(num_columns)
```